src/parcsr_ls/par_cycle.c

```diff
@@ -326,7 +326,7 @@ hypre_BoomerAMGCycle( void              *amg_vdata,
          /* If no coarsening occurred, apply a simple smoother once */
          Aux_U = U_array[level];
          Aux_F = F_array[level];
-         num_sweep = 1;
```

Comment:
@ulrikeyang Can you please comment on this change? This is a special 1-level case. Do we want to apply preconditioner more than once? @PaulMullowney commented on the use case of it as separate preconditioner outside AMG. But do you still use `hypre_BoomerAMGSolve` which calls `hypre_BoomerAMGCycle` to apply?

Reply:
@ulrikeyang Yes, we want to apply the Preconditioner (1 level 2stage GS algorithm) more than once per Krylov iteration. Line 329 was forcing the number of sweeps to 1 every time which was reducing its effectiveness. 

I was still using BoomerAMG but with max_levels=1. Is there another way to do this problem so that this change isn't needed?

Reply:
Yes, I think it's a good idea to allow this, and num_grid_sweeps[0] is not used anywhere., we only use k=1,2,3. We just need to make sure that the default for this continues to be 1, but users will then be able to set it with HYPRE_BoomerAMGSetCycleNumSweeps(solver,ns,0);
So, no extra interface changes needed.

Reply:
@PaulMullowney This is also fine to me. Using preconditioner as smoother of AMG that has only 1 level can fulfill the requirement here, albeit a little tricky. If you want to do it without involving AMG, you need to implement an interface of preconditioner setup and application, and give it to Krylov solvers, which should be very simple. You can follow how it's done for `HYPRE_ParaSailsSetup` and `HYPRE_ParaSailsSolve` for example.